src/binding/python/Attributable.cpp

```diff
@@ -335,6 +375,10 @@ void init_Attributable(py::module &m) {
             return v.getResource();
             // TODO instead of returning lists, return all arrays (ndim > 0) as numpy arrays?
         })
+        .def("get_attribute_type", []( Attributable & attr, std::string const& key ) {
```

Comment:
Ideally we would also return the `numpy.dtype` here for consistency.
```suggestion
        .def("get_attribute_dtype", []( Attributable & attr, std::string const& key ) {
```

Reply:
oh, and we can make this a (read-only) `property` such as `attribute_dtype`

Reply:
Wouldn't a property only work for something without parameters?

Recent commit uses numpy types. There are some workarounds involved, but it's working as long as noone gets the idea to define an attribute of type `array<double,7>` not named `unitDimension`.

I also took the freedom to add a version of `switchType` that only switches over non-vector types, I was annoyed of always having to define a hundred of extra structs. Also, I made two improvements to that version: (1) not having to give the return type any more and (2) making the `action.template operator()<0>(...)` (resp. 1000) optional.

Reply:
Oh, right - didn't see that here.

Probably a cleaner interface on the long term, as in h5py, would be `<an attributable>.attrs` as a key-value store and then we can ask the `.dtype` of the numpy scalars/arrays inside.

Reply:
That would be a breaking interface change though, deprecating our current `Attributable.attributes` call, right?
For now, I've turned this into a readonly property `Attributable.attribute_dtypes` (returning a dictionary: name -> numpy type).